test/unit_tests/test_braket_device.py

```diff
@@ -456,42 +455,28 @@ def test_non_jaqcd_device(name_mock):
 def test_simulator_default_shots():
     """Tests that simulator devices are analytic if ``shots`` is not supplied"""
     dev = _aws_device(wires=2, device_type=AwsDeviceType.SIMULATOR, shots=None)
-    assert dev.shots == 1
-    assert dev.analytic
-
-
-def test_simulator_0_shots():
-    """Tests that simulator devices are analytic if ``shots`` is not supplied"""
-    dev = _aws_device(wires=2, device_type=AwsDeviceType.SIMULATOR, shots=0)
-    assert dev.shots == 1
+    assert dev.shots is None
     assert dev.analytic
 
 
 def test_local_default_shots():
     """Tests that simulator devices are analytic if ``shots`` is not supplied"""
     dev = BraketLocalQubitDevice(wires=2)
-    assert dev.shots == 1
+    assert dev.shots is None
     assert dev.analytic
 
 
-def test_local_0_shots():
-    """Tests that simulator devices are analytic if ``shots`` is not supplied"""
-    dev = BraketLocalQubitDevice(wires=2, shots=0)
-    assert dev.shots == 1
+def test_local_None_shots():
+    """Tests that the simulator devices are analytic if ``shots`` is specified to be `None`."""
+    dev = BraketLocalQubitDevice(wires=2, shots=None)
+    assert dev.shots is None
     assert dev.analytic
 
 
 def test_qpu_default_shots():
```

Comment:
Can you add a case where you can create a QPU device? That seems to be at least one reason why coverage is failing.

Reply:
The coverage is actually failing due to two different lines. Fixed in #63

Reply:
Actually not sure this is the issue anymore, since Cody just added https://github.com/aws/amazon-braket-pennylane-plugin-python/pull/63 to try to fix coverage.